File: java-workshop/part-4-behavior/src/test/java/helper/InMemoryShippingGateway.java
package helper;

import domain.ShippingConfirmation;
import domain.ShippingRequest;
import gateway.ShippingGateway;

public class InMemoryShippingGateway implements ShippingGateway {

    private ShippingConfirmation lastConfirmation;

    @Override
    public ShippingConfirmation createShipment(ShippingRequest request) {
        lastConfirmation = new ShippingConfirmation(
            request.orderId(),
            "TRACK-" + request.orderId(),
            "2024-12-25"
        );
        return lastConfirmation;
    }

    @Override
    public boolean cancelShipment(String trackingNumber) {
        return true;
    }

    public boolean hasCreatedShipment() {
        return lastConfirmation != null;
    }

    public ShippingConfirmation lastConfirmation() {
        return lastConfirmation;
    }
}

File: csharp-workshop/Part4.Behavior.Tests/Helper/InMemoryShippingGateway.cs
using Part4.Behavior.Domain;
using Part4.Behavior.Gateway;

namespace Part4.Behavior.Tests.Helper;

public class InMemoryShippingGateway : IShippingGateway
{
    private ShippingConfirmation? _lastConfirmation;

    public ShippingConfirmation CreateShipment(ShippingRequest request)
    {
        _lastConfirmation = new ShippingConfirmation(
            request.OrderId,
            "TRACK-" + request.OrderId,
            "2024-12-25"
        );
        return _lastConfirmation;
    }

    public bool CancelShipment(string trackingNumber) => true;

    public bool HasCreatedShipment() => _lastConfirmation != null;

    public ShippingConfirmation? LastConfirmation() => _lastConfirmation;
}
